src/NServiceBus.Transport.Msmq/DelayedDelivery/DelayedDeliverySettings.cs

```diff
@@ -0,0 +1,125 @@
+        /// </summary>
+        public TimeSpan TimeToTriggerFetchCircuitBreaker
+        {
+            get => timeToTriggerFetchCircuitBreaker;
+            set
+            {
+                Guard.AgainstNegativeAndZero("value", value);
+                timeToTriggerFetchCircuitBreaker = value;
+            }
+        }
+
+        /// <summary>
+        /// Time to wait before triggering the circuit breaker that monitors the dispatching of due delayed messages to the destination. Defaults to 30 seconds.
+        /// </summary>
+        public TimeSpan TimeToTriggerDispatchCircuitBreaker
+        {
+            get => timeToTriggerDispatchCircuitBreaker;
+            set
+            {
+                Guard.AgainstNegativeAndZero("value", value);
+                timeToTriggerDispatchCircuitBreaker = value;
+            }
+        }
+
+        /// <summary>
+        /// Maximum number of recovery failures per second that triggers the recovery circuit breaker. Recovery attempts are attempts to increment the failure
+        /// counter after a failed dispatch and forwarding messages to the error queue. Defaults to 1/s.
+        /// </summary>
+        public int MaximumRecoveryFailuresPerSecond
+        {
+            get => maximumRecoveryFailuresPerSecond;
+            set
+            {
+                Guard.AgainstNegativeAndZero("value", value);
+                maximumRecoveryFailuresPerSecond = value;
+            }
+        }
+
+        /// <summary>
+        /// Configures delayed delivery.
+        /// </summary>
+        /// <param name="delayedMessageStore">The store to keep delayed messages.</param>
+        /// <param name="retries">Number of retries when trying to forward due delayed messages.</param>
+        /// <param name="timeToTriggerStoreCircuitBreaker">Time to wait before triggering the circuit breaker that monitors the storing of delayed messages in the database. Defaults to 30 seconds.</param>
+        /// <param name="timeToTriggerFetchCircuitBreaker">Time to wait before triggering the circuit breaker that monitors the fetching of due delayed messages from the database. Defaults to 30 seconds.</param>
+        /// <param name="timeToTriggerDispatchCircuitBreaker">Time to wait before triggering the circuit breaker that monitors the dispatching of due delayed messages to the destination. Defaults to 30 seconds.</param>
+        /// <param name="maximumRecoveryFailuresPerSecond">Maximum number of recovery failures per second that triggers the recovery circuit breaker. Defaults to 1/s.</param>
+        public DelayedDeliverySettings(IDelayedMessageStore delayedMessageStore,
+            int retries = 10,
+            TimeSpan? timeToTriggerStoreCircuitBreaker = null,
+            TimeSpan? timeToTriggerFetchCircuitBreaker = null,
+            TimeSpan? timeToTriggerDispatchCircuitBreaker = null,
+            int maximumRecoveryFailuresPerSecond = 1)
```

Comment:
I'm wondering... as these are also public properties and the args optional, should we just remove the args and have the user set them via the properties.

Reply:
👍 for that, it would make this class much simpler.